packages/next/src/server/config.ts

```diff
@@ -252,6 +252,17 @@ function assignDefaults(
     delete userConfig.experimental.dynamicIO
   }
 
+  // Handle migration of experimental.typedRoutes to typedRoutes
+  if (userConfig.experimental?.typedRoutes !== undefined) {
+    // If typedRoutes was not explicitly set by the user, use the typedRoutes value
+    if (userConfig.typedRoutes === undefined) {
+      userConfig.typedRoutes = userConfig.experimental.typedRoutes
+    }
+
+    // Remove the deprecated property
+    delete userConfig.experimental.typedRoutes
+  }
```

Comment:
We should probably use `warnOptionHasBeenMovedOutOfExperimental` like the other flags